fs_attachment_s3_migration/models/ir_attachment.py

```
@@ -0,0 +1,305 @@
+_logger = logging.getLogger(__name__)
+
+# Tautology to include field-linked attachments in searches.
+# Odoo's ir.attachment._search adds ('res_field', '=', False) by default.
+RES_FIELD_DOMAIN = ["|", ("res_field", "=", False), ("res_field", "!=", False)]
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    @api.model
+    def _s3_migration_domain(self, storage_code):
```

Ruff (0.14.10)

Missing return type annotation for private function `_s3_migration_domain`

(ANN202)